wrappers/python/openmm/app/forcefield.py

```diff
@@ -57,11 +57,11 @@ def _getDataDirectories():
     if _dataDirectories is None:
         _dataDirectories = [os.path.join(os.path.dirname(__file__), 'data')]
         try:
-            from pkg_resources import iter_entry_points
-            for entry in iter_entry_points(group='openmm.forcefielddir'):
+            from importlib_metadata import entry_points
+            for entry in entry_points().select(group='openmm.forcefielddir'):
                 _dataDirectories.append(entry.load()())
         except:
-            pass # pkg_resources is not installed
+            pass # importlib_metadata is not installed
```

Comment:
I think from Python 3.8 this is a native module, so you can do `import importlib.metadata`.

The full construct would look more like:

```python
try:
    from importlib import metadata as _metadata
except ModuleNotFoundError:
    import importlib_metadata as _metadata

_metadata.entry_points().select(...)
```

Reply:
It appears the API of the native module has changed with time.  `select()` doesn't exist prior to Python 3.10.  Instead `entry_points()` returns a dict.  `importlib_metadata` also supports the dict interface, but prints a deprecation warning to the console.  Supporting all the different combinations will be a bit of a mess.

Reply:
Does this look right to you?

```python
eps = None
try:
    from importlib.metadata import entry_points
    eps = entry_points()
except:
    try:
        from importlib_metadata import entry_points
        eps = entry_points()
    except:
        pass # importlib_metadata is not installed
if eps is not None:
    if hasattr(eps, 'select'):
        entries = eps.select(group='openmm.forcefielddir')
    else:
        entries = eps['openmm.forcefielddir']
    for entry in entries:
          _dataDirectories.append(entry.load()())
```

Reply:
Or we could just assume ` importlib_metadata` will be installed.  I think it always will be on miniconda?

Reply:
I don't think we can assume it will always be installed (but we can add it to the `run` dependencies). That said, the code proposed above does look solid!

Reply:
I tried the above code, but several of the builds are failing.  I'm going to revert it and just assume `importlib_metadata` is installed.  It generally seems to be anyway, and we can add it as a dependency if we want to make absolutely sure.